### PR TITLE
Layer infobox button fixed

### LINKED ISSUE #9752: Layer hidden infobox button is too big
screen shot 2016-09-14 at 12 53 02

Don't you think @piensaenpixel @urbanphes @josecruz ?

cc @nobuti

### PR DESCRIPTION
**Pull request by MariaCheca**
Fixes #9752 

Changed button style to small and secondary, aligned to the right.

screen shot 2016-09-14 at 16 42 45

CR @matallo please :)

### CODE REVIEW

**Comment by matallo** [comment]
LGTM

**Comment by xavijam** [comment]
Hey, sorry to include myself, but, take care of this:

screen shot 2016-09-14 at 18 04 10

File: lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl [hunk 1]
```diff
@@ -17,8 +17,8 @@
     <% } %>
     <% if (hasButtons) { %>
       <ul class="Infobox-buttons">
-        <li class="Infobox-button js-primary"></li>
-        <li class="Infobox-button Infobox-button--right js-secondary"></li>
+        <li class="Infobox-button js-secondary"></li>
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl]: I feel this is not enough, with this change we can't achieve this layout (already in the UI):
screen shot 2016-09-14 at 19 38 56

Comment (MariaCheca) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl]: That's a good one, with already in the UI do you mean it's already working or jut in the designs?

Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl]: Yep, working AFAIK.

**Comment by MariaCheca** [comment]
@matallo could you review this again? I made several changes after your last review :)

**Comment by xavijam** [comment]
Hey, sorry to "add my self" to the conversation, but I encourage you @MariaCheca to ask @nobuti to review it too, because he implemented the infobox logic.

**Comment by nobuti** [comment]
It's ok, six eyes always better than four. Taking a look, but please @matallo feel free to do the same.

File: lib/assets/javascripts/cartodb3/components/infobox/infobox-factory.js [hunk 2]
```diff
@@ -38,14 +39,16 @@ module.exports = {
...
         type: opts.confirmType || 'link',
-        disabled: opts.confirmDisabled
+        disabled: opts.confirmDisabled,
+        position: opts.confirmPosition
+      },
+      secondAction: {
+        label: opts.cancelLabel,
+        type: opts.cancelType || 'link',
+        disabled: opts.cancelDisabled,
+        position: opts.cancelPosition
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox-factory.js]: 👍  I'd had go with a default position here as fallback, but only because I'm lazy.

File: lib/assets/javascripts/cartodb3/components/infobox/infobox-item-view.js [hunk 3]
```diff
@@ -33,12 +33,14 @@ module.exports = CoreView.extend({
       this._mainLabel = opts.mainAction.label;
       this._mainType = opts.mainAction.type;
       this._mainDisabled = opts.mainAction.disabled;
+      this._mainPosition = opts.mainAction.position;
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox-item-view.js]: if not optional, better to be explicit and throw an error if param is not present.

File: lib/assets/javascripts/cartodb3/components/infobox/infobox-factory.js (hunk 2 above)
Comment (MariaCheca) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox-factory.js]: Good point, I'll add it :)

File: lib/assets/javascripts/cartodb3/components/infobox/infobox-item-view.js (hunk 3 above)
Comment (MariaCheca) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox-item-view.js]: Totally agree!

File: lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl [hunk 4]
```diff
@@ -17,8 +17,8 @@
     <% } %>
     <% if (hasButtons) { %>
       <ul class="Infobox-buttons">
-        <li class="Infobox-button js-primary"></li>
-        <li class="Infobox-button Infobox-button--right js-secondary"></li>
+        <li class="Infobox-button js-leftAction"></li>
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl]: I found this a bit confusing and hard to follow. Here we have `leftAction` and `rightAction`, but in the code we have `mainAction` and `secondAction` too. I feel we are mixing things. I'd probably leave sort of as it was, `left` and `right`  for position, and `js-action`. At the end, a button can have only one action.

Comment (MariaCheca) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl]: Yes, maybe that's a bit confusing. I could change those `.js-[left|right]Action` with `.js-[left|right]Position`.

Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl]: 👍

File: lib/assets/javascripts/cartodb3/components/infobox/infobox-item-view.js (hunk 3 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox-item-view.js]: +1

File: lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl (hunk 4 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/components/infobox/infobox.tpl]: agree, as I would do in CSS I'd relate action to main|second and position to left|right, otherwise it can be confusing

**Comment by MariaCheca** [comment]
Changes applied, thank you both!
